web/src/main/kotlin/com/example/ajaxproject/service/PrivateChatServiceImpl.kt

```diff
@@ -13,79 +13,83 @@ import org.bson.types.ObjectId
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 
 @Service
 class PrivateChatServiceImpl(
     private val privateChatRoomRepository: PrivateChatRoomRepository,
     private val privateChatMessageRepository: PrivateChatMessageRepository,
-    private val userService: UserService,
+    private val userService: UserService
 ) : PrivateChatService {
 
-    companion object {
-        private val logger: Logger = LoggerFactory.getLogger(GroupChatServiceImpl::class.java)
+    override fun createPrivateRoom(senderId: String, recipientId: String): Mono<PrivateChatRoom> {
+        val roomId = roomIdFormat(senderId, recipientId)
+        return Mono.zip(
+            userService.getById(senderId),
+            userService.getById(recipientId)
+        ).flatMap {
+            val sender = it.t1
+            val recipient = it.t2
+            privateChatRoomRepository.findChatRoomById(roomId)
+                .switchIfEmpty(
+                    privateChatRoomRepository.save(
+                        PrivateChatRoom(
+                            id = roomId,
+                            senderId = sender.id,
+                            recipientId = recipient.id
+                        )
+                    )
+                )
+        }
```

Comment:
What is going here:
1. find sender
2. find recipient
3. find chat room
4. if the chat room doesn't exist then save it using sender and recipient values

----

Maybe more correct

1. find chat room
2. if the chat room doesn't exist **then** find sender and recipient. Save to the DB

Reply:
I agree that its better. But still, I will not get rid of requests for users - during creation I will need to check whether such users exist